src/components/LabelGroup/component.tsx

```diff
@@ -13,16 +14,30 @@ class LabelGroup extends BaseComponent <LabelGroupArgs, any> {
 
     attachElement = (nodeElement: HTMLElement, containerElement: any) => {
         if (!nodeElement) return;
+        const childrenErrorMessage = 'A LabelGroup must contain a single PCUI react component as a child';
+        // check that the LabelGroup has a single child
         if (Array.isArray(this.props.children) || !this.props.children) {
-            throw new Error('A LabelGroup must contain a single child react component');
+            throw new Error(childrenErrorMessage);
+        }
+        // casting child as a single ReactElement as we have confirmed it is above
+        const child = this.props.children as ReactElement;
+        const fieldProps = child.props as Record<any, any>;
+        // check if the ReactElement contains an instance of a BaseComponent as its type, confirming it is a PCUI react component
+        if (!((child.type as JSXElementConstructor<any>).prototype instanceof BaseComponent)) {
+            throw new Error(childrenErrorMessage);
+        }
+        // it's safe to cast the ReactElement type as a BaseComponent as we have confirmed it is a BaseComponent above
+        const fieldClass = (child.type as unknown as typeof BaseComponent).ctor;
```

Comment:
I've never seen `as` used twice in an expression. What's that doing exactly?

Reply:
It doesn't look like it is necessary so i'll remove it. I had thought the child.type class couldn't be assigned as a BaseComponent.